(/social-connections) align with dashboard terminology; use partials (#1855)
clerk/clerk-docs: docs/authentication/social-connections/linkedin-oidc.mdx

    }
  ]}
>
  - Use LinkedIn OIDC to authenticate users with OAuth
</TutorialHero>

Enabling OpenID Connect (OIDC) with [LinkedIn](https://learn.microsoft.com/en-us/linkedin/shared/authentication/authentication) allows your users to sign up and sign in to your Clerk application with their LinkedIn account.

## Configure for your development instance

For _development instances_, Clerk uses preconfigured shared OAuth credentials and redirect URIs — no other configuration is needed.

To configure your development instance, follow these steps:

1. In the Clerk Dashboard, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
1. Select the **Add connection** button and select **For all users**.
1. In the **Choose provider** dropdown, select **LinkedIn**.
1. Select **Add connection**.

## Configure for your production instance

In _production instances_, you must provide custom credentials, which involves generating your own **Client ID** and **Client Secret** using your LinkedIn Developer account.

> To make the setup process easier, it's recommended to keep two browser tabs open: one for the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Linked Developer Portal](https://discord.com/developers/applications).

Author: ```suggestion
To make the setup process easier, it's recommended to keep two browser tabs open: one for the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [LinkedIn Developer Portal](https://developer.linkedin.com/).
```

<Steps>
  ### Enable LinkedIn as a social connection

  1. In the Clerk Dashboard, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
  1. Select the **Add connection** button, and select **For all users**.
  1. In the **Choose provider** dropdown, select **LinkedIn**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on.
  1. Save the **Redirect URI** somewhere secure. Keep this modal and page open.

  ### Create a LinkedIn application

  > [!TIP]

>   > If you already have a LinkedIn app you'd like to connect to Clerk, select your app from the [My apps](https://www.linkedin.com/developers/apps?appStatus=active) page in the LinkedIn Developer Portal and proceed to the [next step](#get-your-client-id-and-client-secret) in this tutorial.

Author: ```suggestion
  > If you already have a LinkedIn app you'd like to connect to Clerk, select your app from the [**My apps**](https://www.linkedin.com/developers/apps?appStatus=active) page in the LinkedIn Developer Portal and proceed to the [next step](#get-your-client-id-and-client-secret) in this tutorial.
```

  1. On the homepage of the [LinkedIn Developer Portal](https://developer.linkedin.com/), select **Create app**. You'll be redirected to the **Create an app** form.
  1. Fill out the necessary information. Then, select **Create app**. You'll be redirected to the app's **Products** page.
  1. Select the **Auth** tab.
  1. Under **Application credentials**, save the **Client ID** and **Primary Client Secret** somewhere secure.
  1. Under the **OAuth 2.0 settings** section, next to **Authorized redirect URLs for your app**, select the **Edit** icon.

>   1. Select **+ Add redirect URL** and paste the **Redirect URI** you copied from the Clerk Dashboard.

Author: ```suggestion
  1. Select **Add redirect URL** and paste the **Redirect URL** you copied from the Clerk Dashboard.
  1. Select **Update**.
```

  ### Set the Client ID and Primary Client Secret in the Clerk Dashboard

  1. Navigate back to the Clerk Dashboard where the modal should still be open. Paste the **Client ID** and **Primary Client Secret** values that you saved into the respective fields.
  1. Select **Add connection**.

  > [!NOTE]
  > If the modal or page is no longer open, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page in the Clerk Dashboard. Select the connection. Under **Use custom credentials**, paste the values into their respective fields.

  ### Enable OpenID Connect in your LinkedIn application

  1. Select the **Products** tab.

>   1. Next to **Sign In with LinkedIn using OpenID Connect**, select **Request access**.

Author: ```suggestion
  1. Next to **Sign In with LinkedIn using OpenID Connect**, select **Request access**. A modal will open.
  1. Follow the instructions and select **Request access**.
```
